### PR TITLE
Add debian packaging for the geopm service and geopm runtime

### LINKED ISSUE #2109: As a user of the GEOPM service I would like support for Debian packaging and the Ubuntu distribution
**Is your feature request related to a problem? Please describe.**
Ubuntu is a commonly used distribution that uses the Debian packaging system, but GEOPM does not provide Debian packaging. 

**Describe the solution you'd like**
Convert the geopm-service.spec.in file into a debian package directory that will produce a geopm-service.deb file.  This includes updating the names of all dependencies, and following the requirements of Debian packaging.

**Describe alternatives you've considered**
Source installations could be used, but do not provide a clean uninstall mechanism, and require the user to know how to update systemd.

**Additional context**
Would be nice to have Debian packaging for the base packages in geopm/specs.

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Fixes #2109

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Architecture: any
+Depends: ${misc:Depends},
+         ${shlibs:Depends},
+         libgeopm1 (= ${binary:Version}),
+         libgeopmd1 (= ${binary:Version})
+Description: The geopm runtime package provides many built-in
+ features.  An advanced use case is dynamically coordinating hardware
+ settings across all compute nodes used by an distributed HPC
+ application in response to the application's behavior and resource
+ manager requests.  The dynamic coordination is implemented as a
```
Comment (bgeltz) [inline debian/control]: The rules file disables MPI, so I wonder if we should avoid discussing MPI/HPC use cases here?  Also the hierarchy tree?

File: debian/control [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Package: libgeopm1
+Section: libs
+Architecture: any
+Multi-Arch: same
+Pre-Depends: ${misc:Pre-Depends}
+Depends: ${misc:Depends}, ${shlibs:Depends}
+Description: Library supporting the GEOPM Runtime.  This provides the
+ libgeopm library which provides C and C++ interfaces.
+ .
```
Comment (bgeltz) [inline debian/control:59]: Is this just denoting a blank line?  Or typo?

File: debian/rules [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          --sbindir=$(prefix)/sbin \
+          --datadir=$(prefix)/share \
+	  --mandir=$(prefix)/share/man \
+          --docdir=$(prefix)/share/doc/geopm \
+          --with-bash-completion-dir=/etc/bash_completion.d \
+          --disable-mpi \
+          --disable-openmp \
+          --disable-fortran \
+          --disable-geopmd-local \
+	  --disable-mpi
```
Comment (bgeltz) [inline debian/rules]: Inconsistent indentation.

File: debian/rules [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+override_dh_auto_test:
+	# make check
```
Comment (bgeltz) [inline debian/rules]: ?

**Review by bgeltz — CHANGES_REQUESTED** [review]
I still want to do some local deb builds and testing.

File: debian/control [hunk 5]
```diff
@@ -0,0 +1,84 @@
+Source: geopm-runtime
```
Comment (cmcantalupo) [inline debian/control:1]: Package name changes should be reflected in the RPM based spec files as well.

File: debian/rules (hunk 4 above)
Comment (cmcantalupo) [reply debian/rules]: The test code doesn't currently build on Ubuntu 22

File: debian/rules [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+          --docdir=$(prefix)/share/doc/geopm \
+          --with-bash-completion-dir=/etc/bash_completion.d \
+          --disable-mpi \
+          --disable-openmp \
+          --disable-fortran \
+          --disable-geopmd-local \
+          --disable-mpi
+
+override_dh_auto_test:
+	# make check
```
Comment (bgeltz) [inline debian/rules]:
```suggestion
	# make check # TODO Fix issue #2999
```

File: service/debian/rules [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	test -f configure || ./autogen.sh
+	./configure \
+          --prefix=$(prefix) \
+          --includedir=$(prefix)/include \
+          --datadir=$(prefix)/share \
+          --mandir=$(prefix)/share/man \
+          --libdir=$(prefix)/lib/$(DEB_HOST_MULTIARCH)
+
+override_dh_auto_test:
+	# make check
```
Comment (bgeltz) [inline service/debian/rules]:
```suggestion
	# make check # TODO Fix issue #2999
```

**Review by bgeltz — CHANGES_REQUESTED** [review]

**Review by bgeltz — APPROVED** [review]

**Pull request merged**